server/src/version_provider.ts

```diff
@@ -9,60 +9,61 @@
-
-export class NgVersionProvider {
-  constructor(private readonly probeLocation?: string) {}
-
-  /**
-   * Return the version that is found via the probe location, if provided.
-   */
-  get bundledVersion(): NgVersion|undefined {
-    if (!this.probeLocation) {
-      return;
-    }
-    return this.resolve(this.probeLocation);
+function resolve(packageName: string, paths: string[]): NodeModule|undefined {
+  try {
+    // Here, use native NodeJS require instead of ServerHost.require because
+    // we want the full path of the resolution provided by native
+    // `require.resolve()`, which ServerHost does not provide.
+    const resolvedPath = require.resolve(`${packageName}/package.json`, {paths});
+    const packageJson = require(resolvedPath);
+    return {
+      resolvedPath: path.dirname(resolvedPath),
+      version: packageJson.version,
+    };
+  } catch {
   }
+}
 
-  /**
-   * Return the version that is found via current directory, if any.
-   */
-  get localVersion(): NgVersion|undefined {
-    return this.resolve(process.cwd());
+function minVersion(nodeModule: NodeModule, minMajor: number): boolean {
+  if (!nodeModule.version) {
+    return false;
+  }
+  const [majorStr] = nodeModule.version.split('.');
+  if (!majorStr) {
+    return false;
   }
+  const major = Number(majorStr);
+  if (isNaN(major)) {
+    return false;
+  }
+  return major >= minMajor;
+}
 
-  private resolve(dirName: string): NgVersion|undefined {
-    // Here, use native NodeJS require instead of ServerHost.require because
-    // we want the full path of the resolution provided by native
-    // `require.resolve()`, which ServerHost does not provide.
-    let result: NgVersion|undefined;
-    try {
-      // require.resolve() throws if module resolution fails.
-      const resolutionPath = require.resolve(NGLANGSVC, {
-        paths: [dirName],
-      });
-      if (!resolutionPath) {
-        return;
-      }
-      result = {
-        dirName: path.dirname(resolutionPath),
-        version: undefined,
-      };
-      // require would throw if package.json is not strict JSON
-      const packageJson = require(resolutionPath);
-      if (packageJson && packageJson.version) {
-        result.version = packageJson.version;
-      }
-    } finally {
-      return result;
+/**
+ * Resolve the node module with the specified `packageName` that satisfies
+ * the specified minimum major version.
+ * @param packageName
+ * @param minMajor
+ * @param probeLocations
+ */
+export function resolveWithMinMajor(
+    packageName: string, minMajor: number, probeLocations: string[]): NodeModule {
+  for (const location of probeLocations) {
+    const nodeModule = resolve(packageName, [location]);
+    if (!nodeModule) {
+      continue;
+    }
+    if (minVersion(nodeModule, minMajor)) {
+      return nodeModule;
     }
   }
+  throw new Error(
+      `Failed to resolve '${packageName}' from ${JSON.stringify(probeLocations, null, 2)}`);
```

Comment:
Can you add the minMajor we failed to find in this error message?

Reply:
done.